File: Vesuv.Core.Cpp/server/displayServer.cpp
#include "displayServer.h"

namespace Vesuv::Server {

	using namespace System;
	using namespace Vesuv::Core;


	DisplayServer::DisplayServer() {
		DisplayServer::singleton = this;
	}


	DisplayServer::~DisplayServer() {
		DisplayServer::singleton = nullptr;
	}


	DisplayServer^ DisplayServer::getSingleton() {
		return DisplayServer::singleton;
	}

}

File: Vesuv.Core/Server/DisplayServer.cs
using Vesuv.Core;

namespace Vesuv.Server
{

	public abstract class DisplayServer :
		CoreObject
	{

		private readonly Logger logger;

		public static DisplayServer Singleton { get; private set; }

		protected DisplayServer() {
			Singleton = this;
			this.logger = Logger.GetFor();

			this.logger.Log("Initialize DisplayServer");
		}

		protected override void Dispose(bool disposing) {
			if (disposing) {
				this.logger.Log("Destroy DisplayServer");
				this.logger.Dispose();
				Singleton = null;
			}
			base.Dispose(disposing);
		}

	}

}
